core/src/main/scala/kafka/utils/Log4jController.scala

```diff
@@ -22,44 +22,78 @@ import java.util.Locale
 
 import org.apache.log4j.{Level, LogManager, Logger}
 
+import scala.collection.mutable
+import scala.collection.JavaConverters._
 
-/**
- * An MBean that allows the user to dynamically alter log4j levels at runtime.
- * The companion object contains the singleton instance of this class and
- * registers the MBean. The [[kafka.utils.Logging]] trait forces initialization
- * of the companion object.
- */
-private class Log4jController extends Log4jControllerMBean {
 
-  def getLoggers = {
-    val lst = new util.ArrayList[String]()
-    lst.add("root=" + existingLogger("root").getLevel.toString)
+object Log4jController {
```

Comment:
As far as I can see, this doesn't have to be synchronized because it is a wrapper on log4j which is thread-safe. https://logging.apache.org/log4j/1.2/faq.html#a1.7

Reply:
and static methods look fine to me for same reason.